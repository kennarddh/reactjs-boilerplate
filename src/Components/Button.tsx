import styled from 'styled-components'

import { Props } from './Types'

const Button = styled.button<Partial<Props>>`
	color: ${({ red }) => (red ? '#ff0000' : '#0000ff')};
`

export default Button
