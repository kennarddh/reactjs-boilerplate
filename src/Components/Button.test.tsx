import { render } from '@testing-library/react'

import Button from './Button'

describe('button', () => {
	it('should render button', () => {
		expect.assertions(1)

		const { getByText } = render(<Button>Button</Button>)

		expect(getByText('Button')).toBeInTheDocument()
	})

	it('should snapshot button', () => {
		expect.assertions(1)

		const { asFragment } = render(<Button />)

		expect(asFragment()).toMatchSnapshot()
	})
})
