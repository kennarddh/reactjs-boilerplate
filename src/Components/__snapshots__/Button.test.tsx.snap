// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button should snapshot button 1`] = `
<DocumentFragment>
  .c0 {
  color: #0000ff;
}

<button
    class="c0"
  />
</DocumentFragment>
`;
